reyna/src/com/b2msolutions/reyna/Dispatcher.java

```diff
@@ -52,17 +61,32 @@ protected Result sendMessage(Message message, HttpPost httpPost, HttpClient http
     public static Result canSend(Context context) {
         ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
         NetworkInfo info = connectivityManager.getActiveNetworkInfo();
-        if(info == null || !info.isConnectedOrConnecting()) {
+        if (info == null || !info.isConnectedOrConnecting()) {
             return Result.NOTCONNECTED;
         }
+        int type = info.getType();
+        Preferences preferences = new Preferences(context);
+
+        try {
+            BlackoutTime blackoutTime = new BlackoutTime(context);
+            if (cantSendOnWlanCharging(context, type, preferences, blackoutTime)) return Result.BLACKOUT;
+            if (cantSendOnWlanDischarging(context, type, preferences, blackoutTime)) return Result.BLACKOUT;
+            if (type == ConnectivityManager.TYPE_MOBILE) {
```

Comment:
need to check for all mobile network types

  if (type != ConnectivityManager.TYPE_MOBILE &&		       
                 type != ConnectivityManager.TYPE_MOBILE_DUN &&		
                 type != ConnectivityManager.TYPE_MOBILE_HIPRI &&